src/figuresClasses.ts

```diff
@@ -1,11 +1,80 @@
-export interface Figure {}
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: string = 'triangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side must be greater than 0!');
+    }
+
+    if (a + b <= c || b + c <= a || c + a <= b) {
+      throw new Error(
+        // eslint-disable-next-line max-len
+        'The longest side of the triangle is greater than the sum of the other two',
+      );
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
+}
 
-export class Triangle implements Figure {}
+export class Circle implements Figure {
+  shape: string = 'circle';
 
-export class Circle implements Figure {}
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    private radius: number,
+  ) {
+    if (radius <= 0) {
```

Comment:
Similar to the Triangle class, you don't need to manually reference `radius` in the constructor body. TypeScript automatically creates a property for it when using the `private` access modifier in the constructor parameter.